regparser/web/jobs/views.py

```diff
@@ -91,21 +110,48 @@ def create(self, request, *args, **kwargs):
         return Response(serialized.data, status=status.HTTP_201_CREATED,
                         headers=headers)
 
+
+class JobViewList(mixins.ListModelMixin,
+                  mixins.CreateModelMixin,
+                  generics.GenericAPIView):
+    """
+    Handles the list view for jobs of all types.
+    Should be subclassed along with ``BaseViewList`` for classes handling
+    specific job types.
+    """
+    queryset = PipelineJob.objects.all()
+    renderer_classes = renderer_classes
+    serializer_class = PipelineJobSerializer
+
+    def filter_queryset(self, request, *args, **kwargs):
+        """
+        Overridden in order to get data from the Redis queue as well as the DB.
+
+        Impure
+            Pulls information from the DB and the Redis queue.
+
+        :arg HttpRequest request: the incoming request.
+
+        :rtype: list[PipelineJob]
+        :returns: List of PipelineJob objects.
+        """
+        queryset = super(JobViewList, self).filter_queryset(request, *args,
+                                                            **kwargs)
+        queryset = add_redis_data_to_job_data(queryset)
+        return queryset
```

Comment:
Hm. I don't know the wider implications of this, but `querysets` aren't quite `list`s -- notably, they can be further filtered, and manipulated. Ideally, we would be returning a `queryset` to meet expectations of any downstream code. That said, I don't know what that downstream code _is_. Maybe paging?